test/gleam/function_test.gleam

```diff
@@ -122,3 +122,51 @@ pub fn tap_test() {
   })
   |> should.equal("Thanks Joe & Louis")
 }
+
+pub fn apply1_test() {
+  let fun = fn(x1) { x1 }
```

Comment:
Could we have tests that ensure the order is preserved and the types can be different? Thanks